src/main/java/cash/z/ecc/android/sdk/tool/SaplingParamTool.kt

```diff
@@ -0,0 +1,135 @@
+import kotlinx.coroutines.withContext
+import okio.Okio
+import java.io.File
+
+class SaplingParamTool {
+
+    companion object {
+        /**
+         * Checks the given directory for the output and spending params and calls [fetchParams] if
+         * they're missing.
+         *
+         * @param destinationDir the directory where the params should be stored.
+         */
+        suspend fun ensureParams(destinationDir: String) {
+            var hadError = false
+            arrayOf(
+                ZcashSdk.SPEND_PARAM_FILE_NAME,
+                ZcashSdk.OUTPUT_PARAM_FILE_NAME
+            ).forEach { paramFileName ->
+                if (!File(destinationDir, paramFileName).exists()) {
+                    twig("ERROR: $paramFileName not found at location: $destinationDir")
+                    hadError = true
+                }
+            }
+            if (hadError) {
+                try {
+                    Bush.trunk.twigTask("attempting to download missing params") {
+                        fetchParams(destinationDir)
+                    }
+                } catch (e: Throwable) {
+                    twig("failed to fetch params due to: $e")
+                    throw TransactionEncoderException.MissingParamsException
+                }
+            }
+        }
+
+        /**
+         * Download and store the params into the given directory.
+         *
+         * @param destinationDir the directory where the params will be stored. It's assumed that we
+         * have write access to this directory. Typically, this should be the app's cache directory
+         * because it is not harmful if these files are cleared by the user since they are downloaded
+         * on-demand.
+         */
+        suspend fun fetchParams(destinationDir: String) = withContext(Dispatchers.IO) {
+            val client = createHttpClient()
+            var failureMessage = ""
+            arrayOf(
+                ZcashSdk.SPEND_PARAM_FILE_NAME,
+                ZcashSdk.OUTPUT_PARAM_FILE_NAME
+            ).forEach { paramFileName ->
+                val url = "${ZcashSdk.CLOUD_PARAM_DIR_URL}/$paramFileName"
+                val request = Request.Builder().url(url).build()
+                val response = client.newCall(request).execute()
+                if (response.isSuccessful) {
+                    twig("fetch succeeded")
+                    val file = File(destinationDir, paramFileName)
+                    if (file.parentFile.exists()) {
+                        twig("directory exists!")
+                    } else {
+                        twig("directory did not exist attempting to make it")
+                        file.parentFile.mkdirs()
+                    }
+                    Okio.buffer(Okio.sink(file)).use {
+                        twig("writing to $file")
+                        it.writeAll(response.body().source())
+                    }
+                    twig("fetch succeeded, done writing $paramFileName")
```

Comment:
Directly copying these functions over is exactly the right thing to do for this github issue (which is only concerned with creating the SaplingParamTool file). 

We can improve the logic here as part of https://github.com/zcash/zcash-android-wallet-sdk/issues/203 by wrapping this entire part of the if statement in a `try` block, then updating the failureMessage in the `catch` block.

However, lets save that for a separate commit as part of working on #203. For now, we can just take it one step at a time, as you've already done.

Nice work.